### PR TITLE
Api800/serverhardware

### PR DESCRIPTION
**Pull request by supreethpy**
**Description**
API 800 Support for Server hardware 

**Check List**

- [X] - New functionality includes testing.
- [ ] - All tests pass for go 1.11 + gofmt checks .
- [X] - New functionality has been documented in the README if applicable.
- [X] - New functionality has been thoroughly documented in the examples (please include helpful comments).
- [X] - New endpoints supported are updated in the endpoints-support.md file.
- [] - Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/server_hardware.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		os.Getenv("ONEVIEW_OV_DOMAIN"),
+		os.Getenv("ONEVIEW_OV_ENDPOINT"),
+		false,
+		800)
+
+//	ovVer, _ :=ovc.GetAPIVersion()
+//	fmt.Println(ovVer)
+	//print("Get list of all server hardware resources")
+	println("Get server hardware list by name")
+	time.Sleep(2)
```
Comment (patrickdappollonio) [inline examples/server_hardware.go]: Is there any reason for the sleep?

File: examples/server_hardware.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	//serverName,err := ovc.GetServerHardwareByName("eco-dl360-3-ilo")
+	serverName,_ := ovc.GetServerHardwareByName("0000A66101, bay 4")
+
+	time.Sleep(2)
```
Comment (patrickdappollonio) [inline examples/server_hardware.go]: Same as comment above.

Comment (supreethpy) [reply examples/server_hardware.go]: Updated as per comment .

File: endpoints-support.md [hunk 3]
```diff
@@ -96,8 +96,11 @@
...
 |     **Server Hardware**                                                                                                                          |
-|<sub>/rest/server-hardware</sub>                                                         | GET      | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   |
-|<sub>/rest/server-hardware/{id}</sub>                                                    | GET      | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   | 
+|<sub>/rest/server-hardware</sub>                                                         | GET      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/server-hardware/{id}</sub>                                                    | GET      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/PowerState</sub>   		                                                  | GET      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
```
Comment (jyothis-gireesan) [inline endpoints-support.md]: Better to correct the indentation and spacing to match the file.

File: examples/server_hardware.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	ovc := ClientOV.NewOVClient(
+		os.Getenv("ONEVIEW_OV_USER"),
+		os.Getenv("ONEVIEW_OV_PASSWORD"),
+		os.Getenv("ONEVIEW_OV_DOMAIN"),
+		os.Getenv("ONEVIEW_OV_ENDPOINT"),
+		false,
+		800)
+
+	fmt.Println("Get server hardware list by name")
+	serverName, _ := ovc.GetServerHardwareByName("0000A66101, bay 4")
```
Comment (AnushaY1916) [inline examples/server_hardware.go]: This method returns error as well. So better to capture it and validate it.

**Comment by supreethpy** [comment]
New PR raised due to issue in the branch 

https://github.com/HewlettPackard/oneview-golang/pull/137

**Pull request closed without merging**